src/metabase/driver/sync/fingerprint.clj

```diff
@@ -0,0 +1,21 @@
+(ns metabase.driver.sync.fingerprint
+  "Driver specific behavior related to fingerprinting operations."
+  (:require [metabase.db.metadata-queries :as metadata-queries]
+            [metabase.driver :as driver]))
+
+(defmulti ^:private table-rows-sample
+  "Produces a sample of rows using the given `driver`, from the given `table`'s `fields`.  The default implementation
+  runs a row sampling MBQL query through the regular qp, and passes the `rff` into that, but other drivers may
+  occasionally need different logic.  `opts` is an optional map that may contain additional parameters (see below).
+
+  Note to driver developers: you should almost certainly NOT implement this method.  99% of the time, generating and
+  running a regular MBQL query to fingerprint fields is the right thing to do for sampling rows.
+
+  `opts` parameters:
+  `:truncation-size`: [optional] size to truncate text fields if the driver supports expressions"
+  {:arglists '([driver table fields rff opts]), :style/indent 1}
+  driver/dispatch-on-initialized-driver
+  :hierarchy #'driver/hierarchy)
```

Comment:
A couple of requests:

1. Can you add a blurb about why we made this `^:private`? (e.g. because it's experimental and we're only adding it on a trial basis for now and will decide to keep or remove it in the future). Otherwise in a year or two I'm worried no one will remember the Slack conversation we all had
2. Can you add `:added` metadata? It's still useful to know when we added a new driver multimethod and if we make this public in the future we'll want that info
3. This really shouldn't be `:style/indent 1`. I think the old function this replaced had that metadata but it didn't really make sense there either -- this is a normal function call and not a macro that should have special indentation. Can you remove it?

Reply:
Just to clarify is this in its own namespace just to make it more private? Basically all of the other "main" driver methods are `metabase.driver`, and it's sort of nice to have them all in one place. I know the `:sql` and `:sql-jdbc` ones are spread out across a bunch of namespaces but I'm not sure that's  good thing TBH. If anything I think we should be trying to consolidate namespaces with driver methods rather than creating new ones.

I will definitely make an exception for an `EXPERIMENTAL-DO-NOT-IMPLEMENT` namespace tho.